openapi/ebay.negotiation/client.bal

```diff
@@ -0,0 +1,144 @@
+    # buyers, sellers can use this method (sendOfferToInterestedBuyers) to send an offer to the buyers who are 
+    # interested in the listing. The offer gives buyers the ability to purchase the associated listings at a discounted 
+    # price. For details about how to create seller offers to buyers, see Sending offers to buyers.
+    #
+    # + xEbayCMarketplaceId - The eBay marketplace on which your listings with &quot;eligible&quot; buyers appear. For a 
+    # complete list of supported marketplaces, see Negotiation API requirements and restrictions.
+    # + payload - Send offer to eligible items request.
+    # + return - Success
+    remote isolated function sendOfferToInterestedBuyers(string xEbayCMarketplaceId, CreateOffersRequest payload) 
+                                                        returns SendOfferToInterestedBuyersCollectionResponse|error {
+        string  path = string `/send_offer_to_interested_buyers`;
+        map<any> headerValues = {"X-EBAY-C-MARKETPLACE-ID": xEbayCMarketplaceId};
+        map<string|string[]> accHeaders = getMapForHeaders(headerValues);
+        http:Request request = new;
+        json jsonBody = check payload.cloneWithType(json);
+        request.setPayload(jsonBody);
+        SendOfferToInterestedBuyersCollectionResponse response = check self.clientEp->post(path, request, 
+        headers = accHeaders, targetType=SendOfferToInterestedBuyersCollectionResponse);
+        return response;
+    }
+}
+
+# Generate query path with query parameter.
+#
+# + queryParam - Query parameter map
+# + return - Returns generated Path or error at failure of client initialization
+isolated function  getPathForQueryParam(map<anydata>   queryParam)  returns  string|error {
+    string[] param = [];
+    param[param.length()] = "?";
+    foreach  var [key, value] in  queryParam.entries() {
+        if  value  is  () {
+            _ = queryParam.remove(key);
+        } else {
+            if  string:startsWith( key, "'") {
+                 param[param.length()] = string:substring(key, 1, key.length());
+            } else {
+                param[param.length()] = key;
+            }
+            param[param.length()] = "=";
+            if  value  is  string {
+                string updateV =  check url:encode(value, "UTF-8");
+                param[param.length()] = updateV;
+            } else {
+                param[param.length()] = value.toString();
+            }
+            param[param.length()] = "&";
+        }
+    }
+    _ = param.remove(param.length()-1);
+    if  param.length() ==  1 {
+        _ = param.remove(0);
+    }
+    string restOfPath = string:'join("", ...param);
+    return restOfPath;
+}
+
+# Generate header map for given header values.
+#
+# + headerParam - Headers  map
+# + return - Returns generated map or error at failure of client initialization
+isolated function  getMapForHeaders(map<any>   headerParam)  returns  map<string|string[]> {
```

Comment:
@aneeshafedo hope this generation is fine. `map<any> headerParam` using type `any`